olexandr.vakulinskiy/PR1/practice.rb

```diff
@@ -0,0 +1,18 @@
+require 'faraday'
+
+class Image
+  def download(url, file_name)
+    response = Faraday.get url
+
+    raise ArgumentError, "wrong url, status code #{response.status}" unless response.status == 200
+    raise TypeError, 'file is not an image' unless image?(url[/\.\w{3,4}($|\?)/])
+
+    File.open("#{file_name}#{url[/\.\w{3,4}($|\?)/]}", 'wb') { |fp| fp.write(response.body) }
```

Comment:
Extension will not be valid if provided url contains params like
https://miro.medium.com/max/4000/1*KUy_KKExZrSpBuv9XfyBgA.png?somasda=asd&location=Cherkassy

Reply:
![Screenshot from 2020-12-09 23-39-48](https://user-images.githubusercontent.com/45165843/101691766-dd1b3880-3a77-11eb-8676-8f39e028cc2b.png)
I've been trying to fix this regex, here's the best version, but it matches some whitespaces, what can I do?

Reply:
instead of parsing url u can parse response.content_type

Reply:
Oh, thanks, its very helpful